### PR TITLE
Add Streams API and Storm SQL docs to index page

### PR DESCRIPTION
**Pull request by HeartSaVioR**
Rearrange a bit and add documentations about Streams API and some missing Storm SQL.

### CODE REVIEW

**Comment by ptgoetz** [comment]
+1

File: docs/index.md [hunk 1]
```diff
@@ -31,6 +28,31 @@ Trident is an alternative interface to Storm. It provides exactly-once processin
 * [Trident spouts](Trident-spouts.html)       -- transactional and non-transactional data intake
 * [Trident RAS API](Trident-RAS-API.html)     -- using the Resource Aware Scheduler with Trident.
 
+#### Streams API
+
+Stream APIs is another alternative interface to Storm. It provides a typed API for expressing streaming computations and supports functional style operations.
+
+NOTE: Streams API is an `experimental` feature, and further works might break backward compatibility.
+We're also noticing it via annotating classes with marker interface `@InterfaceStability.Unstable`.
```
Comment (arunmahadevan) [inline docs/index.md]: ~~noticing it~~ notifying it

File: docs/index.md [hunk 2]
```diff
@@ -31,6 +28,31 @@ Trident is an alternative interface to Storm. It provides exactly-once processin
...
+NOTE: Streams API is an `experimental` feature, and further works might break backward compatibility.
+We're also noticing it via annotating classes with marker interface `@InterfaceStability.Unstable`. 
+
+* [Streams API](Stream-API.html)
+
+#### SQL
+
+The Storm SQL integration allows users to run SQL queries over streaming data in Storm.  
+
+NOTE: Storm SQL is an `experimental` feature, so the internal of Storm SQL and supported features are subject to change.
```
Comment (arunmahadevan) [inline docs/index.md]: ~~internal~~ internals

File: docs/index.md [hunk 3]
```diff
@@ -31,6 +28,31 @@ Trident is an alternative interface to Storm. It provides exactly-once processin
...
+We're also noticing it via annotating classes with marker interface `@InterfaceStability.Unstable`. 
+
+* [Streams API](Stream-API.html)
+
+#### SQL
+
+The Storm SQL integration allows users to run SQL queries over streaming data in Storm.  
+
+NOTE: Storm SQL is an `experimental` feature, so the internal of Storm SQL and supported features are subject to change. 
+But small change will not affect the user experience. We will notice/announce the user when breaking UX change is introduced.
```
Comment (arunmahadevan) [inline docs/index.md]: ~~notice/announce~~ notify

File: docs/index.md (hunk 1 above)
Comment (HeartSaVioR) [inline docs/index.md]: Addressed.

File: docs/index.md (hunk 2 above)
Comment (HeartSaVioR) [inline docs/index.md]: Addressed.

File: docs/index.md (hunk 3 above)
Comment (HeartSaVioR) [inline docs/index.md]: Addressed.

**Comment by HeartSaVioR** [comment]
Thanks for reviewing. I addressed all of review comments. I'll merge this.

**Pull request merged**